### PR TITLE
feat: [PLATO-409] Segment integration & consent management

### PR DESCRIPTION
**Pull request by mattvanvoorst-contentful**
**_What will change?_**

**Segment/typewriter tracking**
Added Segment / Typewriter for analytics purposes

More intricate documentation is available through the readme file that was added in the analytics folder, but in a nutshell:
- Added a boilerplate for generating a typewriter.yml file, which is then used to generate a typed typewriter.analytics object.
- Sentry is set uo to report any Segment schema violation
- The configuration of Segment is driven by environment variables
- The typewriter schema is automatically regenerated on various events

**Consent management**
- Added @contentful/experience-consent-manager for Contentful's Osano based consent management
- Built a custom RQ hook that creates a ConsentManager instance and exposes it globally
- Added a little component that injects the custom styling, and adds a button to open the consent settings drawer

Ticket: https://contentful.atlassian.net/browse/PLATO-409
Docs: https://contentful.atlassian.net/wiki/spaces/PROD/pages/4138500311/Templates+-+Segment+tracking

**_Learnings_**
* Typewriter and Segment are reliant on each other, but diverge a lot in types and package versions
* Typewriter's config doesn't really accommodate for different workspaces in Segment. 

**_Testing steps_**
* Follow the instructions to set up the additional configurations in the `analytics/readme` file
* Please check out the branch, run `yarn install` and confirm if the process completes successfully.
* Confirm the `trackingPlans` property has a valid ID in the generates `typewriter.yml` file
* Confirm `yarn typewriter:update` runs successfully
* (_optional_) Attempt to send a typewriter event to Segment.

### CODE REVIEW

**Review by AdrianLThomas — APPROVED** [review]

File: analytics/readme.MD [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+The following environment variables are required for the Typewriter implementation to work.
+
+```dotenv
+NEXT_PUBLIC_SEGMENT_WRITEKEY=
+SEGMENT_TRACKINGPLAN_ID=
+SEGMENT_TYPEWRITER_API_TOKEN=
+```
+
+Where to find the values is documented on https://contentful.atlassian.net/wiki/spaces/PROD/pages/edit-v2/4138500311.
```
Comment (cf-remylenoir) [inline analytics/readme.MD:45]: Not a biggie, but the current link is the page in edit mode.

```suggestion
Where to find the values is documented at https://contentful.atlassian.net/l/cp/1Sh1XmJn.
```

File: src/_ctf-private/ctf-consent-management/useCtfConsent.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const useCtfConsent = (props?: Pick<BaseConfig, 'onConsentChanged' | 'onError'>) =>
+  useQuery({
+    queryKey: ['osano-content'],
+    queryFn: async () => {
+      const manager = createConsentManager({
+        appKey: process.env.NEXT_PUBLIC_OSANO_APP_KEY || '',
+        osano: {
+          configId: process.env.NEXT_PUBLIC_OSANO_CONFIG_ID || '',
+        },
```
Comment (cf-remylenoir) [inline src/_ctf-private/ctf-consent-management/useCtfConsent.ts:12]: Should we handle the errors when the keys are missing? I was tried and this is what happens. If one follows the docs, it should be sufficient.

Screenshot 2023-02-16 at 15 22 02

Are we error handling the `useCtfConsent` the existence of those? What would happen if they are not defined?

**Comment by cf-remylenoir** [comment]
Did we change the plan name since the work started? After generating the `plan.json`, I have a diff:

```diff
- "name": "starter-templates-plato",
+ "name": "starter-templates",
```

File: src/_ctf-private/ctf-consent-management/useCtfConsent.ts (hunk 2 above)
Comment (mattvanvoorst-contentful) [reply src/_ctf-private/ctf-consent-management/useCtfConsent.ts:12]: It shouldn't ever be an issue for our users as the key _should_ always be there, but I'll have a look if we can put a fix on this

**Comment by mattvanvoorst-contentful** [comment]
> Did we change the Segment tracking plan name since this work started? After generating the `plan.json`, I have a diff:
> 
> ```diff
> - "name": "starter-templates-plato",
> + "name": "starter-templates",
> ```
> 
> Asking to know if we need to push this change.

We have yes! Adrian pointed out we should keep the names team agnostic, so we removed it. The generated files are prone to change. and are meant to be in our version control

File: src/_ctf-private/ctf-consent-management/useCtfConsent.ts (hunk 2 above)
Comment (cf-remylenoir) [reply src/_ctf-private/ctf-consent-management/useCtfConsent.ts:12]: That is true, this is a non-blocking comment, feel free to merge this work!

**Review by cf-remylenoir — APPROVED** [review]
Great work, QA'ed with success 👍

**Pull request merged**